cmd/minikube/cmd/service.go

```diff
@@ -74,21 +75,26 @@ var serviceCmd = &cobra.Command{
 			os.Exit(1)
 		}
 
-		var urlString []string
-
-		urlString, err = service.WaitForService(api, namespace, svc,
-			serviceURLTemplate, serviceURLMode, https, wait, interval)
+		urls, err := service.WaitForService(api, namespace, svc, serviceURLTemplate, serviceURLMode, https, wait, interval)
 		if err != nil {
 			exit.WithError("Error opening service", err)
 		}
 
-		if len(urlString) != 0 {
-			out.T(out.Celebrate, "Opening kubernetes service  {{.namespace_name}}/{{.service_name}} in default browser...", out.V{"namespace_name": namespace, "service_name": svc})
+		for _, u := range urls {
+			_, err := url.Parse(u)
```

Comment:
If --format has been passed, will `u` contain that formatted value?  If so, this would imply that `--urls --format {{.IP}}` would generate a warning, when it really shouldn't.  If my understanding is correct, I'd think you'd want the `if serviceURLMode` to come before the `url.Parse(u)` call.

Reply:
Based on some local testing, apparently that's not an issue